src/extension/android/adb.ts

```diff
@@ -104,26 +79,48 @@ export class Adb extends AdbEnhancements {
     /**
      * Sends an intent which launches the main activity of the application.
      */
-    public launchApp(projectRoot: string, packageName: string, debugTarget?: string): Q.Promise<void> {
+    public static launchApp(projectRoot: string, packageName: string, debugTarget?: string): Q.Promise<void> {
         let launchAppCommand = `adb ${debugTarget ? "-s " + debugTarget : ""} shell am start -n ${packageName}/.MainActivity`;
         return new CommandExecutor(projectRoot).execute(launchAppCommand);
     }
 
-    public stopApp(projectRoot: string, packageName: string, debugTarget?: string): Q.Promise<void> {
+    public static stopApp(projectRoot: string, packageName: string, debugTarget?: string): Q.Promise<void> {
         let stopAppCommand = `adb ${debugTarget ? "-s " + debugTarget : ""} shell am force-stop ${packageName}`;
         return new CommandExecutor(projectRoot).execute(stopAppCommand);
     }
 
-    public apiVersion(deviceId: string): Q.Promise<AndroidAPILevel> {
+    public static apiVersion(deviceId: string): Q.Promise<AndroidAPILevel> {
         return this.executeQuery(deviceId, "shell getprop ro.build.version.sdk").then(output =>
             parseInt(output, 10));
     }
 
-    public reverseAdb(deviceId: string, packagerPort: number): Q.Promise<void> {
+    public static reverseAdb(deviceId: string, packagerPort: number): Q.Promise<void> {
         return this.execute(deviceId, `reverse tcp:${packagerPort} tcp:${packagerPort}`);
     }
 
-    private parseConnectedDevices(input: string): IDevice[] {
+    public static showDevMenu(deviceId?: string): Q.Promise<void> {
+        let command = `adb ${deviceId ? "-s " + deviceId : ""} shell input keyevent ${KeyEvents.KEYCODE_MENU}`;
+        return this.commandExecutor.execute(command);
+    }
+
+    public static reloadApp(deviceId?: string): Q.Promise<void> {
+        let commands = [
+            `adb ${deviceId ? "-s " + deviceId : ""} shell input keyevent ${KeyEvents.KEYCODE_MENU}`,
+            `adb ${deviceId ? "-s " + deviceId : ""} shell input keyevent ${KeyEvents.KEYCODE_DPAD_UP}`,
+            `adb ${deviceId ? "-s " + deviceId : ""} shell input keyevent ${KeyEvents.KEYCODE_DPAD_CENTER}`,
+        ];
+
+        return this.executeChain(commands);
```

Comment:
Instead of your own chaining logic you could simply leverage `Array.reduce` (see https://github.com/kriskowal/q#sequences), e.g. 
```js
return commands.reduce((promise, command) => {
    return promise.then(() => this.commandExecutor.execute(command));
}, Q(void 0));
```